### PR TITLE
Modify loading of email white/blacklists

### LINKED ISSUE #9780: pgcleanup complains about disposable_email_whitelist
Error is:

```
ERROR _process_config(): CONFIGURATION ERROR: Can't open supplied whitelist file from path: /gpfs1/data/galaxy_server/galaxy-dev/config/disposable_email_whitelist.conf
```

### PR DESCRIPTION
**Pull request by jdavcs**
Fix #9780 + some refactoring + change of behavior:

Currently, an error is logged when galaxy tries to open a file based on the value of `whitelist_file` config option. This happens on each startup because the option has a default value specified in the schema, so it is always set. I think logging an error is misleading if the user did not set the option, in which case skipping loading the file is expected behavior.

I've considered several options, such as warning if the file is missing or raising an error only if the file is missing and the value is *not* the schema default. The behavior in this current version, I think, will be least surprising for the user: do not provide a default value (so the option will be set only if the user explicitly sets it), and raise (not log) an error if the specified file is not found. 

Specific changes:
- Refactor loading of whitelist and blacklist options: remove duplication
- Expect both files to be in <config_dir> (unless absolute path set). This was the case from the start, when `config/` [was hardcoded into the default value of `blacklist_file`](https://github.com/galaxyproject/galaxy/commit/3b5d8e58376d56a2c210aa043b385f5f1809cd5f#diff-aca5b6c2096078bc08a5a5af7b9b1bdaR270)
- Raise error if option is set but file is not found
- use `strip` instead of `rstrip` for each line (because why not)
- Remove default value from whitelist (least surprise)
- Minor edits to schema wording
- Make example file names consistent (remove "disposable" prefix: makes no sense in whitelist, and unnecessary in blacklist)
- Adjust unit test (value still tested, but not for a default path)

Other considerations:
- Should we provide [a graceful way for handling old settings](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L514)? 
- This is a minor bug with multiple changes to behavior. I don't think we should backport it to 20.05.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/config_schema.yml [hunk 1]
```diff
@@ -930,22 +930,32 @@ mapping:
...
         desc: |
           E-mail domains blacklist is used for filtering out users that are using
-          disposable email address during the registration.  If their address domain
-          matches any domain in the blacklist, they are refused the registration.
+          disposable email addresses at registration.  If their address domain matches any
+          domain on the list, they are refused registration.
 
-          Example value 'config/disposable_email_blacklist.conf'
+          If this option is set, galaxy will expect the file to be at the specified
+          location and will raise an error if the file is not found.
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/config_schema.yml]: Not sure this clarification is needed.

File: lib/galaxy/config/__init__.py [hunk 2]
```diff
@@ -732,6 +716,14 @@ def _process_config(self, kwargs):
                 'filters': ['stack']
             }
 
+    def _load_list_from_file(filepath):
+        try:
+            with open(filepath) as f:
+                return [line.strip() for line in f]
+        except IOError:
+            log.error("CONFIGURATION ERROR: Can't open supplied file: %s", filepath)
+            raise
```
Comment (nsoranzo) [inline lib/galaxy/config/__init__.py]: I would simplify to:
```suggestion
        with open(filepath) as f:
            return [line.strip() for line in f]
```

File: lib/galaxy/webapps/galaxy/config_schema.yml (hunk 1 above)
Comment (jdavcs) [reply lib/galaxy/webapps/galaxy/config_schema.yml]: I wasn't sure either. I'll remove it.

File: lib/galaxy/webapps/galaxy/config_schema.yml [hunk 3]
```diff
@@ -930,22 +930,26 @@ mapping:
...
 
-          Example value 'config/disposable_email_blacklist.conf'
+          Example value 'email_blacklist.conf'
+
+          The value of this option will be resolved with respect to <config_dir>.
 
       whitelist_file:
         type: str
-        default: disposable_email_whitelist.conf
-        path_resolves_to: config_dir
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/config_schema.yml:941]: `path_resolves_to` is not fully used yet, but why remove it?

Comment (jdavcs) [reply lib/galaxy/webapps/galaxy/config_schema.yml:941]: `path_resolves_to` requires a default value. If we specify a default value, the attribute will be set, and an error will be raised (or logged in the current version) when galaxy tries to open that file on startup (that was the bug in #9780). I think the problem with having a default value set for a path is that there is no way to know whether a file is missing because the user did not intend to use it, or because they did, chose to stick with the default, and then forgot to provide the file. In the first case we don't want an error, in the second case we do.

Comment (nsoranzo) [reply lib/galaxy/webapps/galaxy/config_schema.yml:941]: Maybe `path_resolves_to` shouldn't require a default value? I think the only place where a `None` value would be a problem is https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L241-L242 . Maybe these 2 lines could be removed?

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/config_schema.yml (hunk 3 above)
Comment (jdavcs) [reply lib/galaxy/webapps/galaxy/config_schema.yml:941]: Maybe; I'm not sure. There might be corner cases where a default would be needed. I'll check/test this. (I'm mostly off today, so I might not get to it until tomorrow.)

**Comment by ic4f** [comment]
@nsoranzo I will follow-up on decoupling `default` value from `path_resolves_to` in a separate PR (unless I find any issues, in which case I'll ping you).
@dannon thanks for merging!
